File: matlab/tagtools/tagiofuns/strip_quotes.m
function    s = strip_quotes(s)

%     s = strip_quotes(s)
%     Remove bracketing double quotes from string or
%     cell array containing strings.
%
%     Valid: Matlab, Octave
%     markjohnson@st-andrews.ac.uk
%     last modified: 12 July 2017

if nargin<1,
   help strip_quotes
end

if nargin<2,
   nowhite = 0 ;
end

if ischar(s),
   k = find(isspace(s)==0) ;
   s = s(k(1):k(end)) ;
   if length(s)>=2 && all(s([1 end])=='"'),
      s = s(2:end-1) ;
   end
   return
end

[m,n] = size(s) ;
for k=1:m*n,
   ss = s{k} ;
   if ~isempty(ss) && ischar(ss),
      kk = find(isspace(ss)==0) ;
      ss = ss(kk(1):kk(end)) ;
      if length(ss)>=2 && all(ss([1 end])=='"'),
         ss = ss(2:end-1) ;
      end
      s{k} = ss ;
   end
end
return

File: Python/tagiofuns/strip_quotes.py
def strip_quotes(s=None):
    """
    Remove bracketing double quotes from string or list containing strings.
    
    s = strip_quotes(s)

    Valid: Python
    markjohnson@st-andrews.ac.uk; dmwisniewska@gmail.com
    last modified: 07 July 2021

    """

    if not s:
        help(strip_quotes)
        return []
        
    if isinstance(s,str):
        s = s.strip()
        if len(s)>=2 and (s[0]=='"' and s[-1]=='"'):
            s = s[1:-1]
        return s
    
    for k,ss in enumerate(s):
        if ss and isinstance(ss,str):
            ss = ss.strip()
            if len(ss)>=2 and (ss[0]=='"' and ss[-1]=='"'):
                ss = ss[1:-1]
            s[k] = ss
    return s